scripts/validate_datetime.py

```diff
@@ -0,0 +1,61 @@
+"""
+This is pure Python implementation of DateTime
+
+For doctests run following command:
+
+python3 validate_datetime.py
+
+"""
+
+
+import datetime
+import doctest
+import os
+
+
+def parseOptions():
+
+    import optparse
+
+    parser = optparse.OptionParser(usage="-h")
+    parser.add_option("-d", "--difference", type="int")
+    (options, args) = parser.parse_args()
+    return options
+
+
+now = datetime.datetime.now()
+
+
+def subtime(a, b):
+    """(datetime,int) -> datetime
+    Subtract b hours from a datetime.datetime and return the new datetime object
+
+    >>> subtime(datetime.datetime(2013,11,11,11,0),10)
+    datetime.datetime(2013, 11, 11, 1, 0)
+
+    >>> subtime(datetime.datetime(2013,11,11,11,0),24)
+    datetime.datetime(2013, 11, 10, 11, 0)
+
+    >>> subtime(datetime.datetime(2013,11,11,11,0),0)
+    datetime.datetime(2013, 11, 11, 11, 0)
+
+    >>> subtime(datetime.datetime(2013,11,11,11,0),-5)
+    datetime.datetime(2013, 11, 11, 16, 0)
+
```

Comment:
Could you add an example of a failing doctest and also add some check within this function ensuring that the types are correct